pandas/core/reshape/merge.py

```diff
@@ -1106,7 +1107,10 @@ def get_result(self) -> DataFrame:
         join_index, left_indexer, right_indexer = self._get_join_info()
 
         result = self._reindex_and_concat(join_index, left_indexer, right_indexer)
-        result = result.__finalize__(self, method=self._merge_type)
+        result = result.__finalize__(
+            types.SimpleNamespace(input_objs=[self.left, self.right]),
+            method=self._merge_type,
+        )
```

Comment:
Note: I don't really understand why `__finalize__` is called twice on the result, once here with `method=self._merge_type` and once below with `method="merge"`. That looks dubious. If the duplication is needed, we should add a comment.